src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,37 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
+            int length = Interop.GlobalizationNative.ToAscii(flags, unicode, count, null, 0);
```

Comment:
right but if we allocate extra memory in advance we can avoid calling GetAscii twice most of the time.